lib/tessel/deploy.js

```diff
@@ -296,9 +297,89 @@ actions.sendBundle = function(tessel, filepath, opts) {
   });
 };
 
+actions.glob = {};
+actions.glob = {
+  /*
+    A wrapper that allows for stubbing/spying in our tests.
+   */
+  sync: function(pattern, options) {
+    return glob.sync(pattern, options);
+  },
+
+  /*
+    Generate an array of glob pattern rules defined within all files
+    that match the file name provided.
+
+      actionsglob.rules(target, '.tesselignore')
+        -> will compile patterns found in all nested .tesselignore files
+
+      actionsglob.rules(target, '.tesselinclude')
+        -> will compile patterns found in all nested .tesselinclude files
+
+   */
+  rules: function(target, filesContainingGlobPatterns) {
+    var files = actions.glob.sync(target + '/**/*' + filesContainingGlobPatterns, {
```

Comment:
TODO: add comment note about implied path normalization, which is handled by `glob`